File: umami_server/models/usuario.js
const { Schema, model } = require('mongoose');

const UsuarioSchema = Schema({

    nombre: {
        type: String,
        required: true,
        
    },


    email: {
        type: String,
        required: true,
        unique: true
    },
    password: {
        type: String,
        required: true
    },
   

});

UsuarioSchema.method('toJson', function(){
    const {__v, _id, password, ...Object} = this.toObject();
    Object.uid = id;
    return Object;
})


module.exports = model('Usuario', UsuarioSchema );

File: lib/models/usuario.dart
// To parse this JSON data, do
//
//     final usuario = usuarioFromJson(jsonString);

import 'dart:convert';

Usuario usuarioFromJson(String str) => Usuario.fromJson(json.decode(str));

String usuarioToJson(Usuario data) => json.encode(data.toJson());

class Usuario {
  Usuario({
    required this.id,
    required this.nombre,
    required this.email,
    required this.password,
    required this.v,
  });

  String id;
  String nombre;
  String email;
  String password;
  int v;

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        id: json["_id"],
        nombre: json["nombre"],
        email: json["email"],
        password: json["password"],
        v: json["__v"],
      );

  Map<String, dynamic> toJson() => {
        "_id": id,
        "nombre": nombre,
        "email": email,
        "password": password,
        "__v": v,
      };
}
